### ISSUE TITLE
Should call `optimizer.step()` before `lr_scheduler.step()`

### ISSUE THREAD

**Issue by jaxon3062** [ThilinaRajapakse/simpletransformers#1486]
I was using **simpletransformers** in  Colab (default environment), when I start training my model, I get this message telling that optimizer.step() was called before lr_scheduler.step(), and that's not supported after PyTorch 1.1.0.

Error message:
`
/usr/local/lib/python3.8/dist-packages/torch/optim/lr_scheduler.py:138:`` UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate

  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
`

ps. I've read the source code and find it might be wrong here: https://github.com/ThilinaRajapakse/simpletransformers/blob/master/simpletransformers/classification/classification_model.py#L957

**Comment by kassy11** [comment]
@ThilinaRajapakse I came across the same Warning. Does this affect the learning results?
